lighthouse-core/audits/large-javascript-libraries.js

```diff
@@ -0,0 +1,165 @@
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'large-javascript-libraries',
+      title: str_(UIStrings.title),
+      failureTitle: str_(UIStrings.failureTitle),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['Stacks'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @return {LH.Audit.Product}
+   */
+  static audit(artifacts) {
+    /** @type {Array<{original: MinifiedBundlePhobiaLibrary, suggestions: MinifiedBundlePhobiaLibrary[]}>} */
+    const libraryPairings = [];
+    const detectedLibs = artifacts.Stacks.filter(stack => stack.detector === 'js');
+
+    const seenLibraries = new Set();
+
+    for (const detectedLib of detectedLibs) {
+      if (!detectedLib.npm || !libStats[detectedLib.npm]) continue;
+
+      const suggestions = librarySuggestions[detectedLib.npm];
+      if (!suggestions) continue;
+
+      if (seenLibraries.has(detectedLib.npm)) continue;
+      seenLibraries.add(detectedLib.npm);
+
+      let version = 'latest';
+      if (detectedLib.version && libStats[detectedLib.npm].versions[detectedLib.version]) {
+        version = detectedLib.version;
+      }
+
+      const originalLib = libStats[detectedLib.npm].versions[version];
+
+      /** @type {Array<{name: string, repository: string, gzip: number}>} */
+      const smallerSuggestions = [];
+      for (const suggestion of suggestions) {
+        if (libStats[suggestion].versions['latest'].gzip > originalLib.gzip) continue;
+
+        smallerSuggestions.push({
+          name: suggestion,
+          repository: libStats[suggestion].repository,
+          gzip: libStats[suggestion].versions['latest'].gzip,
+        });
+      }
+
+      smallerSuggestions.sort((a, b) => a.gzip - b.gzip);
+      if (smallerSuggestions.length) {
+        libraryPairings.push({
+          original: {
+            gzip: originalLib.gzip,
+            name: detectedLib.npm,
+            repository: libStats[detectedLib.npm].repository,
+          },
+          suggestions: smallerSuggestions,
+        });
+      }
+    }
+
+    /** @type {LH.Audit.Details.Table['items']} */
+    const tableDetails = libraryPairings.map(libraryPairing => {
+      const original = libraryPairing.original;
+      const suggestions = libraryPairing.suggestions;
+      const suggestionItems = suggestions.map(suggestion => {
+        return {
+          suggestion: /** @type {LH.Audit.Details.LinkValue} */ ({
+            text: suggestion.name,
+            url: suggestion.repository,
+            type: 'link',
+          }),
+          transferSize: suggestion.gzip,
+          wastedBytes: original.gzip - suggestion.gzip,
+        };
+      });
+
+      return {
+        name: /** @type {LH.Audit.Details.TableItem} */ {
+          text: original.name,
+          url: original.repository,
+          type: 'link',
+        },
+        transferSize: original.gzip,
+        wastedBytes: 0,
```

Comment:
let's remove `wastedBytes: 0` from the "main item", and use `key: null` for the third heading below, so the "potential savings" value in the main item is just blank instead of 0.